dbms/src/Flash/Mpp/ExchangeReceiver.cpp

```diff
@@ -393,10 +400,10 @@ void ExchangeReceiverBase<RPCContext>::reactor(const std::vector<Request> & asyn
     MPMCQueue<AsyncHandler *> ready_requests(alive_async_connections * 2);
     std::vector<AsyncHandler *> waiting_for_retry_requests;
 
-    std::vector<AsyncRequestHandler<RPCContext>> handlers;
+    std::vector<std::unique_ptr<AsyncHandler>> handlers;
```

Comment:
I can't get what would happen without this change, could you tell me something?

Reply:
In AsyncGrpcExchangePacketReader‘s init function, reader = cluster->rpc_client->sendStreamRequestAsync(xxx), the right part can execute first, and grpc thread sees the reader.

Reply:
yes that's a great job and I've got what happened about this bug.

The only remaining question is why you change the element type of handlers from a value type to a unique_ptr. 

It's ok, but why?

Reply:
mutex field is not support for copy constructor...